### PR TITLE
Support `Microsoft.Data.SqlClient` in S2857.

### LINKED ISSUE #6580: FN S2857: Add support for Microsoft SqlClient Data Provider for SQL Server
### Description

S2857 currently supports System.Data.SqlClient which is deprecated by [Microsoft.Data.SqlClient](https://github.com/dotnet/SqlClient).

> Microsoft.Data.SqlClient is a data provider for Microsoft SQL Server and Azure SQL Database. Now in General Availability, it is a union of the two System.Data.SqlClient components which live independently in .NET Framework and .NET Core. Going forward, support for new SQL Server features will be implemented in Microsoft.Data.SqlClient.

[Source](https://github.com/dotnet/SqlClient)

[Introducing the new Microsoft.Data.SqlClient - .NET Blog](https://devblogs.microsoft.com/dotnet/introducing-the-new-microsoftdatasqlclient/)

See also #6205

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
Closes #6580

### CODE REVIEW

**Comment by martin-strecker-sonarsource** [comment]
Related #6205

File: analyzers/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs [hunk 1]
```diff
@@ -54,6 +54,7 @@ public sealed class SqlKeywordsDelimitedBySpace : SonarDiagnosticAnalyzer
             CSharpSyntaxHelper.BuildQualifiedNameSyntax("System", "Data", "Odbc"),
             CSharpSyntaxHelper.BuildQualifiedNameSyntax("System", "Data", "OracleClient"),
             CSharpSyntaxHelper.BuildQualifiedNameSyntax("Microsoft", "Data", "Sqlite"),
+            CSharpSyntaxHelper.BuildQualifiedNameSyntax("Microsoft", "Data", "SqlClient"),
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs:57]: This needs to be covered by tests.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs (hunk 1 above)
Comment (teo-tsirpanis) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs:56]: Done.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. Thank you @teo-tsirpanis

**Pull request merged**